File: freecodecamp/bonfire/11-chuncky-monkey.rb
def chunk(array, size)
  num_of_groups = array.length / size
  grouped_array = []
  num_of_groups.times { grouped_array.push(array.shift(size)) }
  array.length == 0 ? grouped_array : grouped_array.push(array)
end

p chunk(['a','b','c','d'], 2)
p chunk([0, 1, 2, 3, 4, 5], 3)
p chunk([0, 1, 2, 3, 4, 5], 2)
p chunk([0, 1, 2, 3, 4, 5], 4)

File: freecodecamp/bonfire/11-chuncky-monkey.js
function chunk(arr, size) {
  var groupedArray = [];
  var loops = arr.length / size;
  for (var i = 0; i < loops; i++) {
    var group = [];
    for (var j = 0; j < size; j++) {
      if (arr.length > 0 ) {
        group.push(arr.shift());
      }
    }
    groupedArray.push(group);
  }
  return groupedArray;
}

console.log(chunk(["a", "b", "c", "d"], 2));
console.log(chunk([0, 1, 2, 3, 4, 5], 3));
console.log(chunk([0, 1, 2, 3, 4, 5], 2));
console.log(chunk([0, 1, 2, 3, 4, 5], 4));
